src/handlers.cpp

```diff
@@ -1818,6 +1823,41 @@ void ImpuRegDataTask::on_sar_response(Diameter::Message& rsp)
     SAS::report_event(event);
     _http_rc = HTTP_NOT_FOUND;
   }
+  // An SAA received with this experimental result code may be updating the
+  // wildcarded public identity. If it is, then return to searching the cache
+  // with the new wildcarded public identity and proceed from there. But if it
+  // isn't, treat it as an error.
+  else if (experimental_result_code == DIAMETER_ERROR_IN_ASSIGNMENT_TYPE)
+  {
+    const std::string current_wildcard = (_hss_wildcard.empty() ? _sprout_wildcard : _hss_wildcard);
+    saa.wildcarded_public_identity(_hss_wildcard);
+    if (current_wildcard == _hss_wildcard)
+    {
+      int type = 0;
+      saa.server_assignment_type(type);
+      TRC_INFO("Server-Assignment answer with result code %d and experimental "
```

Comment:
Can the trace statement also include the current_wildcard? (We're not saying that this is definitely a wildcard issue here, but including it in the trace makes it quicker to debug).